app/controllers/RejectedNotificationsController.scala

```diff
@@ -67,6 +67,11 @@ class RejectedNotificationsController @Inject() (
     }
   }
 
+  def addToSessionThenDisplay(id: String, errorReportVersion: String): Action[AnyContent] = (authenticate andThen verifyEmail).async {
+    implicit request =>
+      Future(Redirect(routes.RejectedNotificationsController.displayPage(id)).addingToSession(errorReportView -> errorReportVersion))
```

Comment:
Here you are using a HTTP 303 redirect to essentially just call the `displayPage` method above (causing a second round trip to the browser and server. Could you not just call the `displayPage` method instead (similar to what you are doing on line 52)? :thinking: